Fix spglib wrapper bug (wrong transpose)

jaemolihm opened the pull request:
Closes #539

It seems that this bug (wrong transpose) was added in #496 when moving from spglib_jll to Spglib.jl.

The tests are a bit verbose but I did not have a good idea to organize them.

mfherbst commented:
Thanks @jaemolihm  for the fix and the testcase and absolutely correct!

The annoying issue here is that the spglib python and C interface use *different* conventions for how the lattice vectors are organised (in rows versus in columns) and Spglib.jl seems to be following the python convention of organising it in columns, which is exactly the one we use in DFTK as well.

Could you add a comment about that in the appropriate places you modify? The relevant place in the Spglib.jl docs is 
https://singularitti.github.io/Spglib.jl/stable/api/.

Otherwise this is good to go!

mfherbst commented on src/external/spglib.jl:
```diff
@@ -185,9 +185,9 @@ function get_spglib_lattice(model; to_primitive=false)
-    structure = Spglib.Cell(transpose(model.lattice), spg_positions, spg_numbers)
+    structure = Spglib.Cell(model.lattice, spg_positions, spg_numbers)
     spglib_lattice = Spglib.standardize_cell(structure, to_primitive=to_primitive).lattice
-    Matrix(copy(spglib_lattice'))
+    Matrix(spglib_lattice)
```
```suggestion
    Matrix(Spglib.standardize_cell(structure, to_primitive=to_primitive).lattice)
```

jaemolihm replied:
Done!

antoine-levitt commented:
Oh wow we should have caught that, thanks @jaemolihm! There's a comment at the top of the file so just clarify there that Spglib.jl used the column convention

jaemolihm commented:
I added the comment.
I also added "For example, `model.lattice[:, 1]` is the first real-space lattice vector." in `conventions.md` because it took me some time to understand what "in columns" mean 😂
